okhttp/src/commonJvmAndroid/kotlin/okhttp3/Call.kt

```diff
@@ -87,9 +88,79 @@ interface Call : Cloneable {
    */
   fun timeout(): Timeout
 
+  /**
+   * Returns the tag attached with [type] as a key, or null if no tag is attached with that key.
+   *
+   * The tags on a call are seeded from the [request tags][Request.tag]. This set will grow if new
+   * tags are computed.
+   */
+  fun <T : Any> tag(type: KClass<T>): T?
+
+  /**
+   * Returns the tag attached with [type] as a key, or null if no tag is attached with that key.
+   *
+   * The tags on a call are seeded from the [request tags][Request.tag]. This set will grow if new
+   * tags are computed.
+   */
+  fun <T> tag(type: Class<out T>): T?
+
+  /**
+   * Returns the tag attached with [type] as a key. If it is absent, then [computeIfAbsent] is
+   * called and that value is both inserted and returned.
+   *
+   * If multiple calls to this function are made concurrently with the same [type], multiple values
+   * may be computed. But only one value will be inserted, and that inserted value will be returned
+   * to all callers.
+   *
+   * If computing multiple values is problematic, use an appropriate concurrency mechanism in your
+   * [computeIfAbsent] implementation. No locks are held while calling this function.
+   */
+  fun <T : Any> tag(
+    type: KClass<T>,
+    computeIfAbsent: () -> T,
+  ): T
```

Comment:
We have this law in Canada where packaged goods need to have both French and English on the label. 

<img width="500" height="500" alt="image" src="https://github.com/user-attachments/assets/037fedd5-3948-4028-a103-33309cded051" />

Offering both a Kotlin `KClass` version and a Java `Class` version of these APIs has the same vibes. I don’t want one set of users to feel diminished vs. the other.

Reply:
Oh and if it’s unclear, I believe the french word for Nacho is Nacho. @oldergod can confirm.